src/particles/spacecharge/Gauss3dPush.cpp

```diff
@@ -73,27 +77,35 @@ namespace impactx::particles::spacecharge
             // Simpson's rule integration
 
             // Ex
-            amrex::ParticleReal const fx = f1 / f2x;
+            //amrex::ParticleReal const fx = f1 / f2x;
+            fx = f1 / f2x;
             if (i%2 == 0)
                 sum0ex += 2_prt * fx * h;
             else
                 sum0ex += 4_prt * fx * h;
 
             // Ey
-            amrex::ParticleReal const fy = f1 / f2y;
+            //amrex::ParticleReal const fy = f1 / f2y;
+            fy = f1 / f2y;
             if (i%2 == 0)
                 sum0ey += 2_prt * fy * h;
             else
                 sum0ey += 4_prt * fy * h;
 
             // Ez
-            amrex::ParticleReal const fz = f1 / f2z;
+            //amrex::ParticleReal const fz = f1 / f2z;
+            fz = f1 / f2z;
             if (i%2 == 0)
                 sum0ez += 2_prt * fz * h;
             else
                 sum0ez += 4_prt * fz * h;
         }
 
+        //end point correction
+        sum0ex -= fx*h;
```

Comment:
It confuses me that the `fx` used here is only using the last value of `i=nint` for the calculation of `fx`.

Reply:
Ah, got it. ok end-point of the integration.